@here/harp-datasource-protocol/test/StyleSetEvaluatorTest.ts

```diff
@@ -625,4 +625,76 @@ describe("StyleSetEvaluator", function() {
         assert.deepStrictEqual(decodedTechnique.offset2, ["make-vector", 10, 20, 30]);
         assert.deepStrictEqual(decodedTechnique.offset3, ["make-vector", 10, 20, 30, 40]);
     });
+
+    describe("allow steps in filter conditions", () => {
+        const sse = new StyleSetEvaluator([
+            {
+                id: "new-style-rule",
+                layer: "buildings",
+                when: [
+                    "all",
+                    ["==", ["geometry-type"], "LineString"],
+                    ["step", ["zoom"], false, 15, true]
+                ],
+                technique: "solid-line"
+            },
+            {
+                id: "legacy-style-rule",
+                when: [
+                    "all",
+                    ["==", ["get", "$geometryType"], "line"],
+                    ["==", ["get", "$layer"], "buildings"],
+                    ["step", ["zoom"], false, 15, true]
```

Comment:
This only works for integer zoomlevels right?!
Maybe we can point that out in the documentation.

Reply:
yes, currently it works only with discrete zoom levels. We can make it work also work in general if we re-evaluate the expression in MapView before adding the objects to the scene.